### DISCUSSION TITLE
Unable to obtain Google Services Framework Android ID

### DISCUSSION THREAD

**Discussion by osb99** [tesla-android/issue-tracker#1] [General]
I seem to be struggling to obtain the Google Services Framework Android ID. I enter the 'adb root' command which is successful then I run:
'adb shell 'sqlite3 /data/data/com.google.android.gsf/databases/gservices.db "select * from main where name = \"android_id\";"''

But nothing happens. Am I supposed to enter something specific within that line of code?

<img width="1264" alt="Screenshot 2022-05-07 at 22 41 20" src="https://user-images.githubusercontent.com/69733374/167272531-906f7f14-b463-48f0-bd8b-4ada17fe6728.png">

**Comment by mikegapinski** [comment]
For the command to return anything your Pi needs to be connected to the internet and show the Google Play Protect screen:

![android-setup-google-play](https://user-images.githubusercontent.com/930055/167272841-584714c8-5273-480e-98e3-676d2689c47a.jpg)

You can skip this step of the setup for the time being and just continue without Play Services for a while, it'll work at some point.

Just to confirm that you got the command right copy it directly from Google's website:

https://www.google.com/android/uncertified/?pli=1

adb root
adb shell 'sqlite3 /data/data/com.google.android.gsf/databases/gservices.db \
    "select * from main where name = \"android_id\";"'

**Reply by osb99** [reply]
That's great, thanks!

**Reply by Cooker44** [reply]
Have you been able to connect to Play Store and d/l apps like YouTube, Netflix? Would love to know how to achieve this please

**Reply by CEKSAR23** [reply]
hey, i am new in android world so sorry for f i dont understand basic things. for your method we need to get google services framework id first right? so i cannot get it on phone even I install  many apps like name called android id, my phone id, but even I did factory reset to my phone 5 times its not changed, Still I have error on play store like uncertified. how can i solve it please help  My phone Honor Magic V2 magicos 9 android version
